src/rgw/rgw_http_client_curl.cc

```diff
@@ -0,0 +1,121 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+extern "C" {
+#include <curl/curl.h>
+}
+
+#include "rgw_http_client_curl.h"
+#include <mutex>
+#include <vector>
+
+#include "rgw_common.h"
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+#ifdef WITH_CURL_OPENSSL
+#include <openssl/crypto.h>
+#endif
+
+namespace rgw {
+namespace curl {
+
+static void check_curl()
+{
+#ifndef HAVE_CURL_MULTI_WAIT
+  derr << "WARNING: libcurl doesn't support curl_multi_wait()" << dendl;
+  derr << "WARNING: cross zone / region transfer performance may be affected" << dendl;
+#endif
+}
+
+std::once_flag curl_init_flag;
+
+#ifdef WITH_CURL_OPENSSL
+void init_ssl() {
+  rgw::curl::openssl::init_ssl();
+}
+
+bool fe_inits_ssl(const fe_map_t& m, long& curl_global_flags){
+  for (const auto& kv: m){
+    if (kv.first == "civetweb"){
```

Comment:
just add `|| kv.first == "beast"` - the "ssl_certificate" option is the same there